backend/layers/business/business.py

```diff
@@ -763,6 +818,23 @@ def get_prior_published_dataset_version(self, dataset_version_id: DatasetVersion
         except DatasetVersionNotFoundException:
             return None
 
+    def _delete_from_bucket(self, bucket: str, keys: List[str] = None, prefix: str = None) -> None:
+        try:
+            if keys:
+                self.s3_provider.delete_files(bucket, keys)
+            if prefix:
+                self.s3_provider.delete_recursive(bucket, prefix)
+        except S3DeleteException as e:
+            raise CollectionDeleteException("Attempt to delete public Datasets failed") from e
+
+    def delete_artifacts(self, artifacts: List[DatasetArtifact]) -> None:
+        for artifact in artifacts:
+            logger.info(f"djh artifact is {artifact.uri} {artifact.type}")
+            matches_dict = re.match(r"^s3://(?P<bucket>[^/]+)/((?P<prefix>.*/$)|(?P<key>.*))", artifact.uri).groupdict()
```

Comment:
to avoid all of this regex logic you could factor out this into a comment function 
https://github.com/chanzuckerberg/single-cell-data-portal/blob/ef9c9baf50fe74737d7e9506e9885cbfc8b60131/backend/layers/processing/process_logic.py#L117C5-L124C30

Reply:
You mean to avoid having the regex inlined here? As far as I can tell this specific regex isn't used/needed anywhere else, so factoring it out seems a bit premature to me. If it were several lines I think I would agree but otherwise I'm partial to keep as-is.

Removing the logger comments leftover from dev work... 😅 🤦‍♂️

Reply:
Addressed with business layer unit test `test_delete_artifacts`. Let me know if that suffices.